crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1029,7 +1037,32 @@ pub(crate) fn handle_rename(
     // See https://github.com/microsoft/vscode-languageserver-node/issues/752 for more info
     if !change.file_system_edits.is_empty() && snap.config.will_rename() {
         change.source_file_edits.clear();
+    } else {
```

Comment:
The condition above is `!change.file_system_edits.is_empty()`, so this runs when there are file_system_edits opposed to when its empty, we should just drop the `else` part here

Reply:
We can also move this check *after* the `to_proto::workspace_edit` transformation, then we can lift out the duplicateion of checks by creating a function `fn check_supported_filesystem_edit(config: &Config, op: &ResourceOp) -> Result<()>`. That function will basically check what resource op we have, check if that kind is supported and if not construct and return an appropriate error, otherwise return `Ok(())`

Reply:
Instead of writing a second function I edited what I had but I am also ready to follow your suggestion as well.